math/UsefulMathFunctions.cpp

```diff
@@ -128,14 +144,18 @@ bool isPrime(int n) {
 // After calling this function, prime[i] will be equal 1 if i is prime, 0 otherwise.
 // O(n.log(log(n)))
 bool prime[N];
-void generatePrimes(int n) {
+void generatePrimes(int n)
```

Comment:
What if ```n``` is greater then ```N``` ?

Reply:
The N value that i have used was 1010 but that was just for the purpose of testing you can still modify it .

Reply:
@moadmmh Can you rewrite the functions as I describe